apps/src/code-studio/levels/codeStudioLevels.js

```diff
@@ -0,0 +1,33 @@
+let registeredGetResult = null;
+
+/**
+ * A number of our levels provide a function that can be used to get results.
+ * Previously this was just injected into the global namespace, making it
+ * difficult to track. This makes both the registration and the usage more explicit.
+ */
+export function registerGetResult(getResultFunction=basicGetResult) {
+  if (registeredGetResult) {
+    console.error('already have a getResult function');
+  }
+  registeredGetResult = getResultFunction;
+}
+
+export function getResult() {
+  if (!registeredGetResult) {
+    console.error('No getResult function');
+    return;
```

Comment:
I considered this. However, I wanted to make sure registration was explicit to start, so that if I missed registration anywhere, things fail clearly (instead of just returning the basic version which may or may not be desired).